pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/SegmentCompletionManager.java

```diff
@@ -90,6 +95,8 @@ public SegmentCompletionManager(HelixManager helixManager, PinotLLCRealtimeSegme
     _controllerLeadershipManager = controllerLeadershipManager;
     SegmentCompletionProtocol
         .setMaxSegmentCommitTimeMs(TimeUnit.MILLISECONDS.convert(segmentCommitTimeoutSeconds, TimeUnit.SECONDS));
+    _fsmLocks = new Lock[NUM_FSM_LOCKS];
+    Arrays.fill(_fsmLocks, new ReentrantLock());
```

Comment:
This will fill all locks with the same lock object

Reply:
Thanks for catching that! changed